stan/math/prim/err/check_consistent_sizes.hpp

```diff
@@ -1,150 +1,84 @@
+template <typename T, typename = require_not_vector_t<T>, typename = void>
+inline size_t size_ignore_nonvector(const T&) {
+  return 0;
 }
+}  // namespace internal
 
 /**
- * Check if the dimension of x1, x2, x3, x4 and x5 are consistent.
+ * Check if the dimensions of the inputs are consistent.
  * Consistent size is defined as having the same size if
  * vector-like or being a scalar.
  *
- * @tparam T1 Type of x1
- * @tparam T2 Type of x2
- * @tparam T3 Type of x3
- * @tparam T4 Type of x4
- * @tparam T5 Type of x5
+ * E.g.: check_consistent_sizes("some_function", "x1", x1, "x2", x2, etc.).
  *
- * @param function Function name (for error messages)
- * @param name1 Variable name (for error messages)
- * @param x1 Variable to check for consistent size
- * @param name2 Variable name (for error messages)
- * @param x2 Variable to check for consistent size
- * @param name3 Variable name (for error messages)
- * @param x3 Variable to check for consistent size
- * @param name4 Variable name (for error messages)
- * @param x4 Variable to check for consistent size
- * @param name5 Variable name (for error messages)
- * @param x5 Variable to check for consistent size
- * @throw <code>invalid_argument</code> if sizes are inconsistent
+ * @tparam T1 type of first input
+ * @tparam Ts type of other inputs
+ * @param function function name (for error messages)
+ * @param name1 name of variable corresponding to first input
+ * @param x1 first input
+ * @param names_and_xs more inputs
+ * @throw `invalid_argument` if sizes are inconsistent
  */
-template <typename T1, typename T2, typename T3, typename T4, typename T5>
+template <typename T1, typename... Ts>
 inline void check_consistent_sizes(const char* function, const char* name1,
-                                   const T1& x1, const char* name2,
-                                   const T2& x2, const char* name3,
-                                   const T3& x3, const char* name4,
-                                   const T4& x4, const char* name5,
-                                   const T5& x5) {
-  size_t max_size = std::max(
-      is_vector<T1>::value * stan::math::size(x1),
-      std::max(
-          is_vector<T2>::value * stan::math::size(x2),
-          std::max(is_vector<T3>::value * stan::math::size(x3),
-                   std::max(is_vector<T4>::value * stan::math::size(x4),
-                            is_vector<T5>::value * stan::math::size(x5)))));
-  check_consistent_size(function, name1, x1, max_size);
-  check_consistent_size(function, name2, x2, max_size);
-  check_consistent_size(function, name3, x3, max_size);
-  check_consistent_size(function, name4, x4, max_size);
-  check_consistent_size(function, name5, x5, max_size);
+                                   const T1& x1, const Ts&... names_and_xs) {
+  size_t max_size
+      = std::max({internal::size_ignore_nonvector(x1),
+                  internal::size_ignore_nonvector(names_and_xs)...});
```

Comment:
[comment]
This idiom is super cool.  I didn't realize you could use a brace initializer that way without putting everything in a tuple.

Reply:
This ones new to me as well.